### PR TITLE
Feature add allow batched http requests flag

### PR DESCRIPTION
**Pull request by devkot**
As per https://github.com/apollographql/apollo-server/issues/5686, add the ability to opt out of batches queries in a single HTTP request. This defaults to true for backwards compatibility reasons.

This new feature has a few benefits such as performance as the response in a batch would be bounded by the slowest query. It can also be used as an increased security measure like preventing bad actors from using batching to validate 2FA pin codes by using different permutations.

### CODE REVIEW

File: packages/apollo-server-core/src/runHttpQuery.ts [hunk 1]
```diff
@@ -298,6 +300,18 @@ export async function processHTTPRequest<TContext>(
         parseGraphQLRequest(httpRequest.request, requestParams),
       );
 
+      if (requests.length > 1 && options.allowBatchedHttpRequests === false) {
```
Comment (devkot) [inline packages/apollo-server-core/src/runHttpQuery.ts:303]: I couldn't figure out a nice way of setting defaults in the GraphQLServer options so I've left this conditional naive for now - I'd be keen to improve this

File: packages/apollo-server-core/src/runHttpQuery.ts [hunk 2]
```diff
@@ -298,6 +300,14 @@ export async function processHTTPRequest<TContext>(
         parseGraphQLRequest(httpRequest.request, requestParams),
       );
 
+      if (options.allowBatchedHttpRequests === false) {
```
Comment (glasser) [inline packages/apollo-server-core/src/runHttpQuery.ts:303]: I'm going to move this up a few lines — no use parsing the request if we're going to immediately throw.

**Comment by glasser** [comment]
Thanks @devkot! This should get released this week.

**Comment by glasser** [comment]
This is released in v3.5.0.

**Pull request merged**